packages/sources/gemini/src/transport/reserves.ts

```diff
@@ -0,0 +1,48 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { BaseEndpointTypes } from '../endpoint/reserves'
+
+export interface ResponseSchema {
+  addresses: string[]
+  ethereum_supply: number
+  currency: string
+}
+
+export type HttpTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+export const httpTransport = new HttpTransport<HttpTransportTypes>({
+  prepareRequests: (params, config) => {
+    return params.map((param) => {
+      return {
+        params: [param],
+        request: {
+          baseURL: config.API_ENDPOINT,
+          url: `/v1/tokens/${param.token.toLowerCase()}/reserves`,
+        },
+      }
+    })
+  },
+  parseResponse: (params, response) => {
+    return params.map((param) => {
+      const result = response.data.addresses.map((address) => ({
```

Comment:
Just to be thorough could you add a validation here to make sure `response.data.addresses` exists before we process it?